### PR TITLE
[Discussion] Unify code style

### PR DESCRIPTION
**Pull request by Volcyy**
So, I've went ahead and ran all Python files through [`black`](https://github.com/ambv/black). My motivation behind this is that black enforces a consistent style throughout the project, and with a community project, I think that that's useful.

I'd be very interested for opinions towards this. The only thing I dislike is the wrapping to multiple lines of lines with long comments - but it's easy to fix this by simply putting the comment above, and it's not too common either. Looking forward to your input.

One more thing, it's also possible to use `black` in CI with `black --check` which returns a status code depending on whether or not something would've been changed, useful for not having to check style on PRs ourselves.

### CODE REVIEW

File: bot/cogs/cogs.py [hunk 1]
```diff
@@ -265,9 +300,7 @@ def __init__(self, bot: AutoShardedBot):
 
         embed.colour = Colour.blurple()
         embed.set_author(
-            name="Python Bot (Cogs)",
-            url=GITHUB_URL_BOT,
-            icon_url=BOT_AVATAR_URL
+            name="Python Bot (Cogs)", url=GITHUB_URL_BOT, icon_url=BOT_AVATAR_URL
```
Comment (1mn) [inline bot/cogs/cogs.py]: no thanks.

File: bot/cogs/eval.py [hunk 2]
```diff
@@ -115,9 +117,11 @@ def _format(self, inp, out):  # (str, Any) -> (str, discord.Embed)
                 # Text too long, shorten
                 li = pretty.split("\n")
 
-                pretty = ("\n".join(li[:3]) +  # First 3 lines
-                          "\n ...\n" +  # Ellipsis to indicate removed lines
-                          "\n".join(li[-3:]))  # last 3 lines
+                pretty = (
```
Comment (1mn) [inline bot/cogs/eval.py]: messes up comments. needs a fix.

Also, I kinda like it when comments like these line up. PEP8 requires _at least_ 2 spaces after the code, which seems like a veiled encouragement of lining up consequent inlines.

File: bot/cogs/eval.py [hunk 3]
```diff
@@ -160,11 +164,13 @@ def _format(self, inp, out):  # (str, Any) -> (str, discord.Embed)
             return _
     finally:
         self.env.update(locals())
-""".format(textwrap.indent(code, '            '))
+""".format(
+            textwrap.indent(code, "            ")
+        )
```
Comment (1mn) [inline bot/cogs/eval.py]: no idea why this suddenly is floating out there? This just seems wrong.

File: bot/cogs/events.py [hunk 4]
```diff
@@ -101,36 +108,38 @@ def __init__(self, bot: AutoShardedBot):
...
     async def on_member_update(self, before: Member, after: Member):
-        if before.roles == after.roles and before.name == after.name and before.discriminator == after.discriminator:
+        if (
+            before.roles == after.roles
+            and before.name == after.name
+            and before.discriminator == after.discriminator
+        ):
             return
 
         before_role_names = [role.name for role in before.roles]  # type: List[str]
```
Comment (1mn) [inline bot/cogs/events.py]: can we line up these inline comments too?

File: bot/cogs/hiphopify.py [hunk 5]
```diff
@@ -95,18 +106,13 @@ def __init__(self, bot: AutoShardedBot):
         embed = Embed()
         embed.colour = Colour.blurple()
 
-        params = {
-            "user_id": str(member.id),
-            "duration": duration
-        }
+        params = {"user_id": str(member.id), "duration": duration}
```
Comment (1mn) [inline bot/cogs/hiphopify.py]: doesn't seem like this is an improvement.

File: bot/cogs/math.py [hunk 6]
```diff
@@ -69,18 +78,18 @@ def __init__(self, bot):
                 return
 
             # Send LaTeX to website to get image
-            data = {
-                "latex": parsed,
-                "res": 300,
-                "color": 808080
-            }
+            data = {"latex": parsed, "res": 300, "color": 808080}
```
Comment (1mn) [inline bot/cogs/math.py]: again - I like this multilined.

File: bot/cogs/math.py [hunk 7]
```diff
@@ -69,18 +78,18 @@ def __init__(self, bot):
...
 
             async with self.bot.http_session.post(LATEX_URL, data=data) as resp:
                 html = await resp.text()
 
-            name = search(r'hist\.request\.basename = "(?P<url>[^"]+)"', html).group('url')
+            name = search(r'hist\.request\.basename = "(?P<url>[^"]+)"', html).group(
```
Comment (1mn) [inline bot/cogs/math.py]: this is just weird.

File: bot/cogs/math.py [hunk 8]
```diff
@@ -109,18 +118,18 @@ def __init__(self, bot):
                 return
 
             # Send LaTeX to website to get image
-            data = {
-                "latex": parsed,
-                "res": 300,
-                "color": 808080
-            }
+            data = {"latex": parsed, "res": 300, "color": 808080}
```
Comment (1mn) [inline bot/cogs/math.py]: I also liked this better multilined.

File: bot/cogs/math.py [hunk 9]
```diff
@@ -109,18 +118,18 @@ def __init__(self, bot):
...
 
             async with self.bot.http_session.post(LATEX_URL, data=data) as resp:
                 html = await resp.text()
 
-            name = search(r'hist\.request\.basename = "(?P<url>[^"]+)"', html).group('url')
+            name = search(r'hist\.request\.basename = "(?P<url>[^"]+)"', html).group(
```
Comment (1mn) [inline bot/cogs/math.py]: the fuck

File: bot/formatter.py [hunk 10]
```diff
@@ -61,10 +64,14 @@ class <cog2>:
...
             cog_string = f" from {self.command.cog_name}" if self.command.cog_name else ""
 
-            log.trace(f"Help command is on specific command {self.command.name}{cog_string}.")
+            log.trace(
+                f"Help command is on specific command {self.command.name}{cog_string}."
+            )
 
             # strip the command off bot. and ()
-            stripped_command = self.command.name.replace(HELP_PREFIX, "").replace("()", "")
+            stripped_command = self.command.name.replace(HELP_PREFIX, "").replace(
```
Comment (1mn) [inline bot/formatter.py]: this is also super weird.

File: bot/pagination.py [hunk 11]
```diff
@@ -74,7 +73,10 @@ def add_line(self, line='', *, empty=False):
             The line was too big for the current :attr:`max_size`.
         """
         if len(line) > self.max_size - len(self.prefix) - 2:
-            raise RuntimeError('Line exceeds maximum page size %s' % (self.max_size - len(self.prefix) - 2))
+            raise RuntimeError(
+                "Line exceeds maximum page size %s"
```
Comment (1mn) [inline bot/pagination.py]: why are we using c-style % interpolation?

File: bot/pagination.py [hunk 12]
```diff
@@ -129,26 +141,31 @@ def event_check(reaction_: Reaction, user_: Member):
 
             no_restrictions = (
                 # Pagination is not restricted
-                not restrict_to_user or
+                not restrict_to_user
+                or
                 # The reaction was by a whitelisted user
```
Comment (1mn) [inline bot/pagination.py]: naw, this part doesn't work for me. it'd be better to have an inline comment, I think. or something else. just find any other solution. we can't have that lonely `or`

File: bot/cogs/cogs.py [hunk 13]
```diff
@@ -177,14 +199,19 @@ def __init__(self, bot: AutoShardedBot):
...
+                f"{ctx.author} requested we reload the '{cog}' cog, but that cog doesn't exist."
+            )
             embed.description = f"Unknown cog: {cog}"
 
         if full_cog:
             if full_cog == "*":
                 all_cogs = [
-                    f"bot.cogs.{fn[:-3]}" for fn in os.listdir("bot/cogs")
-                    if os.path.isfile(f"bot/cogs/{fn}") and fn.endswith(".py") and "_" not in fn
+                    f"bot.cogs.{fn[:-3]}"
```
[the remark this answers is not in the document]
Comment (gdude2002) [reply bot/cogs/cogs.py]: :eyes:

[the remark this answers is not in the document]
Comment (1mn) [reply bot/cogs/cogs.py]: 😁

File: bot/cogs/bot.py [hunk 14]
```diff
@@ -8,10 +8,19 @@
 from dulwich.repo import Repo
 
 from bot.constants import (
-    ADMIN_ROLE, BOT_AVATAR_URL, BOT_CHANNEL,
-    DEVTEST_CHANNEL, HELP1_CHANNEL, HELP2_CHANNEL,
-    HELP3_CHANNEL, HELP4_CHANNEL, MODERATOR_ROLE, OWNER_ROLE,
-    PYTHON_CHANNEL, PYTHON_GUILD, VERIFIED_ROLE
+    ADMIN_ROLE,
```
Comment (1mn) [inline bot/cogs/bot.py]: this is still kinda insane.

File: bot/cogs/bot.py [hunk 15]
```diff
@@ -63,11 +73,7 @@ def __init__(self, bot: AutoShardedBot):
         embed.add_field(name="Total Users", value=str(len(self.bot.get_guild(PYTHON_GUILD).members)))
         embed.add_field(name="Git SHA", value=str(sha)[:7])
 
-        embed.set_author(
-            name="Python Bot",
-            url="https://github.com/discord-python/bot",
-            icon_url=BOT_AVATAR_URL
-        )
+        embed.set_author(name="Python Bot", url="https://github.com/discord-python/bot", icon_url=BOT_AVATAR_URL)
```
Comment (1mn) [inline bot/cogs/bot.py]: I still prefer the old one here. whenever we list things like this, it's nice to multiline it.

File: bot/cogs/clickup.py [hunk 16]
```diff
@@ -205,35 +209,27 @@ def __init__(self, bot: AutoShardedBot):
...
-                      f"error: {result['err']}")
-            embed = Embed(
-                colour=Colour.red(),
-                description=f"`{result['ECODE']}`: {result['err']}"
+            log.error(
+                "ClickUp responded to the team request with an error!\n"
+                f"error code: '{result['ECODE']}'\n"
+                f"error: {result['err']}"
             )
+            embed = Embed(colour=Colour.red(), description=f"`{result['ECODE']}`: {result['err']}")
```
Comment (1mn) [inline bot/cogs/clickup.py]: not an improvement

File: bot/cogs/clickup.py [hunk 17]
```diff
@@ -205,35 +209,27 @@ def __init__(self, bot: AutoShardedBot):
...
-                colour=Colour.blurple()
-            )
+            embed = Embed(colour=Colour.blurple())
 
             for member in result["team"]["members"]:
-                embed.add_field(
-                    name=member["user"]["username"],
-                    value=member["user"]["id"]
-                )
+                embed.add_field(name=member["user"]["username"], value=member["user"]["id"])
```
Comment (1mn) [inline bot/cogs/clickup.py]: prefer the old one.

File: bot/cogs/clickup.py [hunk 18]
```diff
@@ -273,15 +264,12 @@ def __init__(self, bot: AutoShardedBot):
 
                 lists = "\n".join(lists)
 
-                embed.add_field(
-                    name=f"{project['name']} ({project['id']})",
-                    value=lists
-                )
+                embed.add_field(name=f"{project['name']} ({project['id']})", value=lists)
```
Comment (1mn) [inline bot/cogs/clickup.py]: I like this multilined.

File: bot/cogs/cogs.py [hunk 19]
```diff
@@ -5,9 +5,15 @@
 from discord.ext.commands import AutoShardedBot, Context, command
 
 from bot.constants import (
-    ADMIN_ROLE, BOT_AVATAR_URL, DEVOPS_ROLE, GITHUB_URL_BOT,
-    GREEN_CHEVRON, MODERATOR_ROLE, OWNER_ROLE, RED_CHEVRON,
-    WHITE_CHEVRON
+    ADMIN_ROLE,
```
Comment (1mn) [inline bot/cogs/cogs.py]: insane

File: bot/cogs/cogs.py [hunk 20]
```diff
@@ -51,11 +57,7 @@ def __init__(self, bot: AutoShardedBot):
         embed = Embed()
         embed.colour = Colour.red()
 
-        embed.set_author(
-            name="Python Bot (Cogs)",
-            url=GITHUB_URL_BOT,
-            icon_url=BOT_AVATAR_URL
-        )
+        embed.set_author(name="Python Bot (Cogs)", url=GITHUB_URL_BOT, icon_url=BOT_AVATAR_URL)
```
Comment (1mn) [inline bot/cogs/cogs.py]: prefer the old way

File: bot/cogs/cogs.py [hunk 21]
```diff
@@ -108,11 +116,7 @@ def __init__(self, bot: AutoShardedBot):
...
```
Comment (1mn) [inline bot/cogs/cogs.py]: old way

File: bot/cogs/cogs.py [hunk 22]
```diff
@@ -163,11 +169,7 @@ def __init__(self, bot: AutoShardedBot):
...
```
Comment (1mn) [inline bot/cogs/cogs.py]: aaand old way.

File: bot/cogs/cogs.py [hunk 23]
```diff
@@ -183,7 +185,8 @@ def __init__(self, bot: AutoShardedBot):
         if full_cog:
             if full_cog == "*":
                 all_cogs = [
-                    f"bot.cogs.{fn[:-3]}" for fn in os.listdir("bot/cogs")
+                    f"bot.cogs.{fn[:-3]}"
+                    for fn in os.listdir("bot/cogs")
                     if os.path.isfile(f"bot/cogs/{fn}") and fn.endswith(".py") and "_" not in fn
```
Comment (1mn) [inline bot/cogs/cogs.py]: I mean, if we're gonna do this then this line should be split into three lines too.

File: bot/cogs/cogs.py [hunk 24]
```diff
@@ -209,10 +212,7 @@ def __init__(self, bot: AutoShardedBot):
                     else:
                         loaded += 1
 
-                lines = [
-                    "**All cogs reloaded**",
-                    f"**Unloaded**: {unloaded} / **Loaded**: {loaded}"
-                ]
+                lines = ["**All cogs reloaded**", f"**Unloaded**: {unloaded} / **Loaded**: {loaded}"]
```
Comment (1mn) [inline bot/cogs/cogs.py]: prefer multiline for all lists with more than one item.

File: bot/cogs/cogs.py [hunk 25]
```diff
@@ -264,11 +265,7 @@ def __init__(self, bot: AutoShardedBot):
         cogs = {}
 
         embed.colour = Colour.blurple()
-        embed.set_author(
-            name="Python Bot (Cogs)",
-            url=GITHUB_URL_BOT,
-            icon_url=BOT_AVATAR_URL
-        )
+        embed.set_author(name="Python Bot (Cogs)", url=GITHUB_URL_BOT, icon_url=BOT_AVATAR_URL)
```
Comment (1mn) [inline bot/cogs/cogs.py]: prefer multiline for all embed calls like this.

File: bot/cogs/deployment.py [hunk 26]
```diff
@@ -62,17 +62,12 @@ def __init__(self, bot: AutoShardedBot):
         response = await self.bot.http_session.get(STATUS_URL)
         data = await response.json()
 
-        embed = Embed(
-            title="Service status",
-            color=Colour.blurple()
-        )
+        embed = Embed(title="Service status", color=Colour.blurple())
```
Comment (1mn) [inline bot/cogs/deployment.py]: prefer multiline for all embed calls like this.

File: bot/cogs/deployment.py [hunk 27]
```diff
@@ -62,17 +62,12 @@ def __init__(self, bot: AutoShardedBot):
...
 
         for obj in data:
             key, value = list(obj.items())[0]
 
-            embed.add_field(
-                name=key, value=value, inline=True
-            )
+            embed.add_field(name=key, value=value, inline=True)
```
Comment (1mn) [inline bot/cogs/deployment.py]: this wasn't even multiline enough for my liking in the first place. I would prefer this on five lines.

File: bot/cogs/eval.py [hunk 28]
```diff
@@ -181,10 +184,12 @@ def _format(self, inp, out):  # (str, Any) -> (str, discord.Embed)
...
-        if not re.search(  # Check if it's an expression
-                r"^(return|import|for|while|def|class|"
-                r"from|exit|[a-zA-Z0-9]+\s*=)", code, re.M) and len(
-                    code.split("\n")) == 1:
+        if (
+            not re.search(  # Check if it's an expression
+                r"^(return|import|for|while|def|class|" r"from|exit|[a-zA-Z0-9]+\s*=)", code, re.M
+            )
+            and len(code.split("\n")) == 1
+        ):
```
Comment (1mn) [inline bot/cogs/eval.py]: this if is pretty hideous both before and after the change. perhaps we can refactor it to something altogether better.

File: bot/cogs/events.py [hunk 29]
```diff
@@ -101,19 +104,13 @@ def __init__(self, bot: AutoShardedBot):
...
-                    title="User roles updated"
-                )
+                embed = Embed(title="User roles updated")
 
                 for key, value in done.items():
                     if value:
-                        embed.add_field(
-                            name=key.title(), value=str(value)
-                        )
+                        embed.add_field(name=key.title(), value=str(value))
```
Comment (1mn) [inline bot/cogs/events.py]: should be 4 lines

File: bot/cogs/events.py [hunk 30]
```diff
@@ -125,12 +122,9 @@ def __init__(self, bot: AutoShardedBot):
...
         log.debug(f"{before.display_name} roles changing from {before_role_names} to {after_role_names}")
 
-        await self.send_updated_users({
-            "user_id": str(after.id),
-            "roles": role_ids,
-            "username": after.name,
-            "discriminator": after.discriminator
-        })
+        await self.send_updated_users(
+            {"user_id": str(after.id), "roles": role_ids, "username": after.name, "discriminator": after.discriminator}
```
Comment (1mn) [inline bot/cogs/events.py]: consistent style here would be

```py
await self.send_updated_users(
    {
        "user_id": str(after.id), 
        "roles": role_ids, 
        "username": after.name, 
        "discriminator": after.discriminator
    }
)
```

File: bot/cogs/fun.py [hunk 31]
```diff
@@ -5,11 +5,7 @@
 
 from bot.constants import BOT_CHANNEL
 
-RESPONSES = {
-    "_pokes {us}_": "_Pokes {them}_",
-    "_eats {us}_": "_Tastes slimy and snake-like_",
-    "_pets {us}_": "_Purrs_"
-}
+RESPONSES = {"_pokes {us}_": "_Pokes {them}_", "_eats {us}_": "_Tastes slimy and snake-like_", "_pets {us}_": "_Purrs_"}
```
Comment (1mn) [inline bot/cogs/fun.py]: better before it was changed.

File: bot/cogs/hiphopify.py [hunk 32]
```diff
@@ -6,9 +6,14 @@
 from discord.ext.commands import AutoShardedBot, Context, command
 
 from bot.constants import (
-    ADMIN_ROLE, MODERATOR_ROLE, MOD_LOG_CHANNEL,
-    NEGATIVE_REPLIES, OWNER_ROLE, POSITIVE_REPLIES,
-    SITE_API_HIPHOPIFY_URL, SITE_API_KEY
+    ADMIN_ROLE,
```
Comment (1mn) [inline bot/cogs/hiphopify.py]: nutzo

File: bot/cogs/hiphopify.py [hunk 33]
```diff
@@ -42,9 +47,7 @@ def __init__(self, bot: AutoShardedBot):
         )
 
         response = await self.bot.http_session.get(
-            SITE_API_HIPHOPIFY_URL,
-            headers=self.headers,
-            params={"user_id": str(before.id)}
+            SITE_API_HIPHOPIFY_URL, headers=self.headers, params={"user_id": str(before.id)}
```
Comment (1mn) [inline bot/cogs/hiphopify.py]: better before.

File: bot/cogs/hiphopify.py [hunk 34]
```diff
@@ -88,26 +91,18 @@ def __init__(self, bot: AutoShardedBot):
         """
 
         log.debug(
-            f"Attempting to hiphopify {member.display_name} for {duration}. "
-            f"forced_nick is set to {forced_nick}."
+            f"Attempting to hiphopify {member.display_name} for {duration}. " f"forced_nick is set to {forced_nick}."
```
Comment (1mn) [inline bot/cogs/hiphopify.py]: certainly not an improvement. kind of a weird thing to do.

File: bot/cogs/hiphopify.py [hunk 35]
```diff
@@ -88,26 +91,18 @@ def __init__(self, bot: AutoShardedBot):
...
 
         if forced_nick:
             params["forced_nick"] = forced_nick
 
-        response = await self.bot.http_session.post(
-            SITE_API_HIPHOPIFY_URL,
-            headers=self.headers,
-            json=params
-        )
+        response = await self.bot.http_session.post(SITE_API_HIPHOPIFY_URL, headers=self.headers, json=params)
```
Comment (1mn) [inline bot/cogs/hiphopify.py]: not an improvement.

File: bot/cogs/hiphopify.py [hunk 36]
```diff
@@ -167,9 +162,7 @@ def __init__(self, bot: AutoShardedBot):
         embed.colour = Colour.blurple()
 
         response = await self.bot.http_session.delete(
-            SITE_API_HIPHOPIFY_URL,
-            headers=self.headers,
-            json={"user_id": str(member.id)}
+            SITE_API_HIPHOPIFY_URL, headers=self.headers, json={"user_id": str(member.id)}
```
Comment (1mn) [inline bot/cogs/hiphopify.py]: ugh.

File: bot/cogs/hiphopify.py [hunk 37]
```diff
@@ -180,10 +173,7 @@ def __init__(self, bot: AutoShardedBot):
             embed.colour = Colour.red()
             embed.title = random.choice(NEGATIVE_REPLIES)
             embed.description = response.get("error_message")
-            log.warning(
-                f"Error encountered when trying to unhiphopify {member.display_name}:\n"
-                f"{response}"
-            )
+            log.warning(f"Error encountered when trying to unhiphopify {member.display_name}:\n" f"{response}")
```
Comment (1mn) [inline bot/cogs/hiphopify.py]: I prefer the way this used to be.

File: bot/cogs/math.py [hunk 38]
```diff
@@ -24,15 +24,20 @@
...
+        [  # noqa: B603,S603
+            sys.executable,
+            "-c",
+            "import sys,sympy;from sympy.parsing.sympy_parser import parse_expr;" f"print(sympy.latex({code_}))",
+            sympy_code,
+        ],
+        env={},  # Disable environment variables for security
+        stdout=PIPE,
+        stderr=STDOUT,
+    )  # reroute all to stdout
```
Comment (1mn) [inline bot/cogs/math.py]: comment in wrong place.

File: bot/cogs/math.py [hunk 39]
```diff
@@ -24,15 +24,20 @@
...
-                    sys.executable, "-c",
-                    "import sys,sympy;from sympy.parsing.sympy_parser import parse_expr;"
-                    f"print(sympy.latex({code_}))", sympy_code
-                 ], env={},  # Disable environment variables for security
-                 stdout=PIPE, stderr=STDOUT)  # reroute all to stdout
-
-    for _ in range(timeout*4):  # Check if done every .25 seconds for `timeout` seconds
-        await asyncio.sleep(1/4)
+    proc = Popen(
+        [  # noqa: B603,S603
```
Comment (1mn) [inline bot/cogs/math.py]: this probably breaks flake8, noqa is on the wrong line.

File: bot/cogs/tags.py [hunk 40]
```diff
@@ -3,32 +3,33 @@
...
-    AutoShardedBot, Context,
-    Converter, command
-)
+from discord.ext.commands import AutoShardedBot, Context, Converter, command
 
 from bot.constants import (
-    ADMIN_ROLE, BOT_COMMANDS_CHANNEL, DEVTEST_CHANNEL,
-    HELPERS_CHANNEL, MODERATOR_ROLE, OWNER_ROLE,
-    SITE_API_KEY, SITE_API_TAGS_URL, TAG_COOLDOWN
+    ADMIN_ROLE,
```
Comment (1mn) [inline bot/cogs/tags.py]: gdude doesn't love this

File: bot/cogs/tags.py [hunk 41]
```diff
@@ -3,32 +3,33 @@
...
 from bot.pagination import LinePaginator
 
 log = logging.getLogger(__name__)
 
-TEST_CHANNELS = (
-    DEVTEST_CHANNEL,
-    BOT_COMMANDS_CHANNEL,
-    HELPERS_CHANNEL
-)
+TEST_CHANNELS = (DEVTEST_CHANNEL, BOT_COMMANDS_CHANNEL, HELPERS_CHANNEL)
```
Comment (1mn) [inline bot/cogs/tags.py]: whyyyy

**Comment by Volcyy** [comment]
I agree that some of the changes are fairly weird, and having an automated program mess with things like `bot/constants.py` or long imports with `from x import (y, z, a, b, c)` is kinda bound for terror. Regarding multi lining, it probably only operates based on line length, sometimes resulting in weird things like
```py
my_long_function_call('does').some_stuff(
    True
)
```
which just looks weird. I also agree with the comments being super weird sometimes. I'll close it for now, maybe we can find a better way to get a more consistent code style throughout the project at some point.

**Comment by 1mn** [comment]
the idea of unifying code style is a great one. and I wouldn't mind if we enforced it strictly - but if we're going to do that, we need to absolutely 100% agree on what that code style should be. if we could find a tool sort of like Black but which allowed us to customize it to our liking ... but I doubt such a tool exists.

**Pull request closed without merging**